[KOGITO-2317] - Unify Kogito Quarkus and Spring Images (#348)
kiegroup/kogito-images: README.md

The Kogito Builder Images are responsible for building the project with Apache Maven and generate the binary that will 
be used by the Kogito Runtime images to run the Kogito application.

The current available Kogito Builder image is:

* [quay.io/kiegroup/kogito-builder](https://quay.io/kiegroup/kogito-builder)

> The builder image supports building application based on Springboot and Quarkus. To define your runtime specify the `RUNTIME_TYPE` enviornment variable. If var is not defined it defaults to `quarkus`

Reviewer: ```suggestion
The builder image supports building applications based on Spring Boot and Quarkus. To define your runtime, specify the `RUNTIME_TYPE` environment variable. If var is not defined, it defaults to `quarkus`.
```

> When `RUNTIME_TYPE` quarkus is chosen the Builder Image allows you to create native image using GraalVM which allows you to have lightweight and fast applications ready to run in the Cloud.

Reviewer: ```suggestion
When `RUNTIME_TYPE` quarkus is chosen, the Builder Image allows you to create a native image using GraalVM, which allows you to have lightweight and fast applications ready to run in the Cloud.
```

The Kogito Builder Image is equipped with the following components:

 * GraalVM 20.2.0-java11
 * OpenJDK 11.0.6
 * Maven 3.6.2
 

> For more information about what is installed on this image take a look [here](kogito-builder-overrides.yaml) in the

Reviewer: ```suggestion
For more information about what is installed on this image, take a look [here](kogito-builder-overrides.yaml) in the
```

**modules.install** section. 

#### Kogito Builder Image usage

This image contains a helper option to better understand how to it:

```bash
$ docker run -it quay.io/kiegroup/kogito-builder:latest /home/kogito/kogito-app-launch.sh -h
```

> By default, quarkus is selected as runtime and a normal java build will be performed. To perform a native build just set the **NATIVE** build environment variable to **true**.

Reviewer: ```suggestion
By default, quarkus is selected as runtime, and a normal java build will be performed. To perform a native build, just set the **NATIVE** build environment variable to **true**.
```

See the next topic for an example.
 

#### Kogito Builder Image example

##### Builder Image Example with Quarkus

> In this example, let's use a simple application based on quarkus and available in the [Kogito Examples](https://github.com/kiegroup/kogito-examples)

Reviewer: ```suggestion
In this example, let's use a simple application based on Quarkus that is available in the [Kogito Examples](https://github.com/kiegroup/kogito-examples)
```

```

##### Builder Image Example with Springboot

> In this example, let's use a simple application based on springboot available in the [Kogito Examples](https://github.com/kiegroup/kogito-examples)

Reviewer: ```suggestion
In this example, let's use a simple application based on Spring Boot that is available in the [Kogito Examples](https://github.com/kiegroup/kogito-examples)
```

> Upon build, these assets will be copied to a generated maven project and built with Maven to produce a runnable binary. Default value of group id is "com.company", artifact id is "project" and version is "1.0-SNAPSHOT". To provide custom value we need to set the **PROJECT_GROUP_ID**, **PROJECT_ARTIFACT_ID** and **PROJECT_VERSION**.

Reviewer 2: for asset, there is an error and this needs to be moved to `Builder image`.
Can we move it with this PR ?

Author: Not sure if I understood correctly, do we need to move `Assets only` section to builder image? and if so is there anything else that needs to be done?

Reviewer 2: yes, it would be great to move that `Assets` part in Builder image chapter as it is a builder thing and not a runtime thing :)

#### Kogito Runtime JVM Image

This Kogito Runtime Image contains only the needed files to execute a pre built Kogito application and a JRE.

> The Image can run application based on Quarkus or Springboot. User can define `RUNTIME_TYPE` enviornment variable to switch between the two.

Reviewer: ```suggestion
The Image can run an application based on Quarkus or Springboot. Users can define `RUNTIME_TYPE` environment variable to switch between the two.
```

#repare the container logs the following message:
Order has been created Order[12345] with assigned approver JOHN
```
**Runtime Image example with springboot**
Let's try, here, the *process-springboot-example*: 

```bash
$ mvn clean package
```

> A uberjar file has been generated into the **target directory**. 
> Let's use this uberjar to perform the build:

Reviewer: ```suggestion
An uberjar file has been generated into the **target directory**. 
Let's use this uberjar to perform the build:
```

Order has been created Order[12345] with assigned approver JOHN
```

#### Kogito Runtime Native Image

This Kogito Runtime Image contains only the needed files to execute a pre built Kogito application.

##### Kogito Runtime Native Image Usage

This image contains a helper option to better understand how to it:

```bash
docker run -it quay.io/kiegroup/kogito-runtime-native:latest /home/kogito/kogito-app-launch.sh -h
```

##### Kogito Runtime Native Image Example

> For this example let's use the same than the previous one (process-quarkus-example). 
> But this time, let's perform a native build:

Reviewer: ```suggestion
For this example, let's use the same as the previous one (process-quarkus-example). 
But this time, let's perform a native build:
```

# installing the imagestream on the current namespace
$ oc create -f https://raw.githubusercontent.com/kiegroup/kogito-images/0.16.0/kogito-imagestream.yaml
imagestream.image.openshift.io/kogito-runtime-native created
imagestream.image.openshift.io/kogito-runtime-jvm created
imagestream.image.openshift.io/kogito-builder created
imagestream.image.openshift.io/kogito-data-index-infinispan created
imagestream.image.openshift.io/kogito-data-index-mongodb created
imagestream.image.openshift.io/kogito-trusty created
imagestream.image.openshift.io/kogito-jobs-service created
imagestream.image.openshift.io/kogito-management-console created

# performing a new build

> $ oc new-build --name=rules-quarkus-helloworld-builder --image-stream=kogito-builder:0.9.0 \ 

Reviewer 2: I guess we should update the version of the image here or use `latest`

Author: Updated to use latest on master branch, m,aybe when we branch and bump-version we can change the version here as well?
